test/types/defineComponent.spec.ts

```diff
@@ -105,6 +112,24 @@ describe('defineComponent', () => {
     expect.assertions(3);
   });
 
+  it('custom props type inferred from PropType', () => {
```

Comment:
this test is passing on `master` branch  🤔

Reply:
I guess this is because the behavior prior TS 3.9 because `PropsType` includes `{}` type which can be assigned to `{ user?: User }`. 🤔 

https://devblogs.microsoft.com/typescript/announcing-typescript-3-9-rc/
> Generally, an intersection type like A & B is assignable to C if either A or B is assignable to C

Once I updated TS to 3.9, the test fails on master.